verifiers/internal/gcb/verifier.go

```diff
@@ -126,3 +125,12 @@ func (v *GCBVerifier) VerifyImage(ctx context.Context,
 	}
 	return content, builderID, nil
 }
+
+// VerifyImageProvenanceRepo verifies provenance for an OCI image.
+func (v *GCBVerifier) VerifyImageProvenanceRepo(ctx context.Context,
```

Comment:
Let's re-use the `VerifyImage()`. Adding an additional parameter will be a breaking change, and that's why you're using a different unction, correct? Let' s keep it simple and just add a parameter to the options.ProvenanceOpts structure. This should make your changes simpler too.


I know retrofitting the provenance repo name in ProvenanceOpts is not as clean as having a dedicated option, but that's a problem with the original API design. We're going to fix this eventually and redesign it with variadic options. @ramonpetgrave64

Reply:
Sure. I was also leaning toward adding ProvenanceOpts struct but it didn't seem to fit from underlying verification and expected params POV. Since refactoring this later is on the table, i will dump in there and submit new changes!! Thx @laurentsimon

Reply:
@laurentsimon Added `provenanceRepository` to `provenanceOpts` struct and reused `verifyImage`